src/ImageSharp/Formats/Gif/GifEncoderCore.cs

```diff
@@ -347,5 +388,61 @@ private void WriteImageData<TPixel>(QuantizedFrame<TPixel> image, Stream stream)
                 encoder.Encode(stream);
             }
         }
+
+        /// <summary>
+        /// Makes pixels that are repeated transparent.
+        /// </summary>
+        /// <param name="renderedFrame">The currently rendered frame.</param>
+        /// <param name="sourceFrame">The source frame.</param>
+        /// <param name="cutFrame">
+        /// The output frame. Pixels will be made transparent if the source color is the same as the rendered color
+        /// and get the same color as in the <paramref name="sourceFrame"/> if it is not.
+        /// </param>
+        private void DoCutRepeatedPixels<TPixel>(Buffer2D<TPixel> renderedFrame, ImageFrame<TPixel> sourceFrame, ImageFrame<TPixel> cutFrame)
```

Comment:
Might actually be nice to put these methods and the memory management in a class (or partial class) to keep it out of the core encoder. This kind of stuff makes it harder to see what's in the GIF spec and what's extra. Also cleaner if more optimizations like this are added later.

Reply:
I think they're ok in this class. It appears to me that what you're doing actually conforms to the specification no?